wa/instruments/delay.py

```diff
@@ -0,0 +1,231 @@
+        Parameter('fixed_between_jobs', kind=int, default=None,
+                  global_alias='fixed_delay_between_jobs',
+                  aliases=['fixed_between_iterations'],
+                  description="""
+                  How long to sleep (in seconds) before starting each
+                  new job.
+
+                  .. note:: This cannot be specified at the same time as
+                            ``temperature_between_jobs``
+                  """),
+        Parameter('fixed_before_start', kind=int, default=None,
+                  global_alias='fixed_delay_before_start',
+                  description="""
+                  How long to sleep (in seconds) after setup for
+                  an iteration has been performed but before running the
+                  workload.
+
+                  .. note:: This cannot be specified at the same time as
+                            ``temperature_before_start``
+                  """),
+        Parameter('temperature_before_start', kind=int, default=None,
+                  global_alias='thermal_threshold_before_start',
+                  description="""
+                  Temperature (in device-specific units) the
+                  device must cool down to just before the actual workload
+                  execution (after setup has been performed).
+                  .. note:: This cannot be specified at the same time as
+                            ``fixed_between_jobs``
+                  """),
+        Parameter('active_cooling', kind=bool, default=False,
+                  description="""
+                  This instrument supports an active cooling
+                  solution while waiting for the device temperature to drop to
+                  the threshold. If you wish to use this feature please ensure
+                  the relevant module is installed on the device.
+                  """),
+    ]
+
+    active_cooling_modules = ['mbed-fan', 'odroidxu3-fan']
+
+    def initialize(self, context):
+        if self.active_cooling:
+            self.cooling = self._discover_cooling_module()
+            if not self.cooling:
+                msg = 'Cooling module not found on target. Please install one of the following modules: {}'
+                raise InstrumentError(msg.format(self.active_cooling_modules))
+
+        if self.temperature_between_jobs == 0:
+            temp = self.target.read_int(self.temperature_file)
+            self.logger.debug('Setting temperature threshold between jobs to {}'.format(temp))
+            self.temperature_between_jobs = temp
+        if self.temperature_between_specs == 0:
+            temp = self.target.read_int(self.temperature_file)
+            msg = 'Setting temperature threshold between workload specs to {}'
+            self.logger.debug(msg.format(temp))
+            self.temperature_between_specs = temp
+
+        signal.connect(self.on_job_change, signal.BEFORE_JOB, priority=CallbackPriority.EXTREMELY_LOW)
```

Comment:
There is a standard map for `signal.BEFORE_JOB` -- just rename `on_job_change` to `before_job`.